### PR TITLE
Show detailed error for captive portal presense

### PR DESCRIPTION
**Pull request by ShivamPokhriyal**
cross-request: https://github.com/dimagi/commcare-core/pull/889

### CODE REVIEW

**Comment by ShivamPokhriyal** [comment]
@damagatchi retest this please

File: app/src/org/commcare/utils/FormUploadUtil.java [hunk 1]
```diff
@@ -183,6 +184,9 @@ private static FormUploadResult submitEntity(List<MultipartBody.Part> parts, Str
             Logger.log(LogTypes.TYPE_ERROR_CONFIG_STRUCTURE,
                     "Encountered PlainTextPasswordException while submission: Sending password over HTTP");
             return FormUploadResult.AUTH_OVER_HTTP;
+        } catch (CaptivePortalRedirectException e) {
+            e.printStackTrace();
+            return FormUploadResult.CAPTIVE_PORTAL;
```
Comment (shubham1g5) [inline app/src/org/commcare/utils/FormUploadUtil.java:189]: we should add a log here as well

**Comment by shubham1g5** [comment]
@ShivamPokhriyal The build is failing because of the foece push, if you clear the workspace on the builder job and rebuild the job, it should work. 

Also wanted to understand what is it that's making you force push commits frequently ?

**Comment by ShivamPokhriyal** [comment]
> @ShivamPokhriyal The build is failing because of the force push, if you clear the workspace on the builder job and rebuild the job, it should work.

Ohh, will take a look at that. 
> Also wanted to understand what is it that's making you force push commits frequently ?
Umm, I guess I usually force push in 2 situations, when I've to rebase to master, or when there is a very small change(changing just 1 line like deleting an import).

**Comment by shubham1g5** [comment]
> when I've to rebase to master

Wondering how are you rebasing. it sounds weird to me if you had to force push because of a rebase or merge. 

> or when there is a very small change(changing just 1 line like deleting an import).

will be good to refrain from it if a simple push works as It's harder to know the new changes after last review when a branch get force pushed, causing reviews to take longer.

**Comment by ctsims** [comment]
@damagatchi retest this please

**Comment by ctsims** [comment]
@ShivamPokhriyal I can check this out to see if it works, but looks like the build is busted?

**Comment by ctsims** [comment]
Nevermind, retest fixed it!

**Pull request merged**